app/src/ui/delete-branch/delete-branch-dialog.tsx

```diff
@@ -95,14 +99,17 @@ export class DeleteBranch extends React.Component<
   }
 
   private deleteBranch = async () => {
-    const { dispatcher, repository, branch, onDeleted } = this.props
+    const { dispatcher, repository, branch } = this.props
+
+    this.setState({ isDeleting: true })
+
     await dispatcher.deleteBranch(
       repository,
       branch,
       this.state.includeRemoteBranch
     )
-    onDeleted(repository)
+    this.props.onDeleted(repository)
 
-    return dispatcher.closePopup()
+    await dispatcher.closePopup()
```

Comment:
The `Button.onClick` callback is a void-returning function, so I updated this to respect that signature rather than returning a `Promise<void>` that wasn't being `await`-ed.